contract_comm/currency/currency.go

```diff
@@ -164,8 +164,8 @@ func Cmp(val1 *big.Int, currency1 *common.Address, val2 *big.Int, currency2 *com
 		return val1.Cmp(val2)
 	}
 
-	exchangeRate1, err1 := getExchangeRate(currency1)
-	exchangeRate2, err2 := getExchangeRate(currency2)
+	exchangeRate1, err1 := memoizedGetExchangeRate(currency1)
```

Comment:
Just to confirm, the *only* place `Cmp` is used is in comparing gas prices for tx inclusion, right? Is there anywhere else this is used that is not safe to memoize?

Reply:
It's used in the tx pool as well to compare newer transactions. It's not use anywhere in consensus (though it should theoretically be safe to use in consensus)